### ISSUE TITLE
Allow Simpler Fetch-Artifact Dependencies

### ISSUE THREAD

**Issue by steve-gray** [gocd/gocd#3185]
##### Issue Type

- Feature proposal
- Feature enhancement

##### Summary

At the moment when specifying a dependency from other pipelines, you must specify the fully qualified path to the specific dependency pipelines. For example:
 
   - actually-wanted/child/sub-child

Due to to the way GoCD fan-in/out works, a pipeline only triggers when all inputs to it are based upon the same input material. In short, even if actually-wanted is referenced in multiple places upstream, the child and sub-child pipelines would _**not**_ trigger unless all upstreams were in agreement.

The reason this matters is because if you are building pipeline templates for complex tasks (i.e. deployment to an environment) and want to chain them in sequence dev->qa->staging->prod, it means that you can't re-use the pipeline templates as you get further down, and worse still, you can't change the shape of the workflow because everywhere has the material full paths baked in, and doing basically anything at this point (i.e. breaking apart a chain to add a new step in the middle) becomes painful or requires basically doing big-bang cruise xml changes.

In short, proposed: allow simpler direct references to upstream pipelines in Fetch Artifact commands, without needing to qualify the path.

**Comment by arvindsv** [comment]
I understand the need and what you're saying. I have one concern: Despite fan-in, it is possible to force upstreams to not be in agreement. That's when this becomes a problem (and not reproducible, depending on the path that was taken). Thoughts? Maybe from you and/or others?

**Comment by steve-gray** [comment]
Perhaps forcing a pipeline run with conflicting materials should just produce a warning, telling the user that they're breaking the fan-out/in principles, and that they won't get downstream automatic scheduling once people enforce the rules:

"You are forcing a pipeline run with materials that have conflicting upstream revisions of the following: [<pipeline-names here>]. Downstream pipelines may not execute automatically and may require manual execution to use any outputs from this operation".

Alternatively, if two inputs use different versions of a common ancestor, notify the user that the latest will be used consistently for that execution and/or all downstreams.
